### PR TITLE
ci(udeps): upgrade actions/checkout to v3

### PR DESCRIPTION
**Pull request by oscard0m**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: .github/workflows/udeps.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
       bundler: ${{ steps.filter.outputs.bundler }}
       cli: ${{ steps.filter.outputs.cli }}
     steps:
-      - uses: actions/checkout@v2
```
Comment (lucasfernog) [inline .github/workflows/udeps.yml:36]: We're still using v2 in several places though

Comment (oscard0m) [reply .github/workflows/udeps.yml:36]: Do you want me to apply the migration to `v3` to all the workflows in the same PR? I found it convinient to do it in separate PRs.

**Comment by oscard0m** [comment]
I'm confused. Why is this PR closed and labeled as spam?

**Comment by lucasfernog** [comment]
It does feel like you're trying to spam PRs for hacktoberfest though, there's 24 files with checkout@v2 and 33 occurrences, so you're suggesting 24 PRs (or even 33) for such a small change? This would even waste our CI though which is already not perfect.

**Comment by FabianLars** [comment]
We initially wanted to just mark it as invalid (or simply close it) because this pr is against the [spirit of hacktoberfest](https://hacktoberfest.com/participation/), but because of what Lucas said, and the fact that you already opened like 100 similar PRs we went with the spam label instead because this really does look fishy to be completely honest.

**Comment by oscard0m** [comment]
> It does feel like you're trying to spam PRs for hacktoberfest though, there's 24 files with checkout@v2 and 33 occurrences, so you're suggesting 24 PRs (or even 33) for such a small change? 

I'm sorry this is being perceived as spam. If it's felt like that I'm happy to close them. 

My intention was to help projects participating in Hacktoberfest to get updated with the latest version of some `actions/`, something we tend to forget as maintainers. I'm also happy to help set up [Renovate](https://github.com/renovatebot/renovate), which in my experience works quite good if you want a more automatic solution. I can even create a GitHub Discussion if you are up to talking about that.

My intention was to suggest an upgrade per each file in `workflows/`, in case there was any kind of problem, it would be more granular to revert but, if you want to do the bump in the same PR, I'm happy to do that as well (it's easier to open one PR than N PRs for me as well).

> This would even waste our CI though which is already not perfect.

I'm sorry I was not aware you were using a self-hosted(?) solution for CI, I just assumed you were using the free Tier of GitHub and that opening a few PRs is not a problem. I would be happy to suggest an update in https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines to remark this as well.

**Comment by oscard0m** [comment]
> We initially wanted to just mark it as invalid (or simply close it) because this pr is against the [spirit of hacktoberfest](https://hacktoberfest.com/participation/)

Re-reading the _Values_ it's not clear to me why these contributions are going against the spirit of Hacktoberfest, to be honest. Still, I can close all the PRs I opened in tauri and apologize for the noise these contributions caused to you (and the costs on CI, I was not aware of that part).

> , but because of what Lucas said, and the fact that you already opened like 100 similar PRs we went with the spam label instead because this really does look fishy to be completely honest.

I posted in my previous comment my idea and intentions regarding my change. I don't see it as fishy but I guess it depends on the angle you see it.

So... if you want I can close my PR's if that is what you want.

In any case, thanks for your time checking this and I'm so sorry for this feeling of spammy contributions but it was not my intention at all. I just wanted to help and add my small grain of sand.

**Pull request closed without merging**